src/main/java/chess/dao/ChessDBService.java

```diff
@@ -0,0 +1,60 @@
+package chess.dao;
+
+import chess.domain.board.ChessBoard;
+import chess.domain.board.GameInformation;
+import chess.domain.piece.Piece;
+import chess.domain.position.Position;
+import chess.dto.ChessGameComponentDto;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ChessDBService {
+    private static final int NEW_GAME_COMMAND = 0;
+    private static final int SOURCE_POSITION_INDEX = 0;
+    private static final int TARGET_POSITION_INDEX = 1;
+
+    private final GameInformationDao gameInformationDao;
+    private final ChessGameDao chessGameDao;
+
+    public ChessDBService(GameInformationDao gameInformationDao, ChessGameDao chessGameDao) {
+        this.gameInformationDao = gameInformationDao;
+        this.chessGameDao = chessGameDao;
+    }
+
+    public List<GameInformation> getAllGameInformation() {
+        return gameInformationDao.findAll();
+    }
+
+    public void removeFinishedGame(ChessBoard chessBoard) {
+        gameInformationDao.remove(chessBoard.getGameId());
+    }
+
+    public void updateChessBoard(List<Position> movedPath, GameInformation gameInformation) {
+        chessGameDao.update(movedPath.get(SOURCE_POSITION_INDEX), movedPath.get(TARGET_POSITION_INDEX));
+        gameInformationDao.updateTurn(gameInformation);
+    }
+
+    public ChessBoard findChessBoard(int gameId) {
+        if (gameId == NEW_GAME_COMMAND) {
+            return createNewChessBoard();
+        }
+        return getSavedChessBoard(gameId);
+    }
+
+    private ChessBoard createNewChessBoard() {
+        gameInformationDao.create();
+        GameInformation gameInformation = gameInformationDao.findLatestGame();
```

Comment:
도메인과 DB가 따로 놀고 있는것 같아요
```suggestion
        GameInformation gameInformation = gameInformationDao.create(new GameInformation());
```
이런 방향으로 한번 작성해보면 좋을것 같습니다! 도메인에서 id를 어떻게 관리할지도 고민해보면 좋겠네요!

Reply:
생성된 도메인 혹은 수정된 도메인을 통해 데이터를 저장, 업데이트하는 방식으로 수정했습니다.
이 과정에서 자동으로 부여되는 game_information.id를 게임 실행에 사용하기 보다 사용자가 지정하는 “name”을 사용하는것이 사용성 측면에서 나을 것 같아서 게임 이름을 갖도록 수정하였습니다. (만약 기존과 같이 고유번호로 관리하고자 한다면 생성될 고유 번호를 먼저 찾고, 해당 번호로 new GameInformation(고유번호) 를 생성한 뒤 ChessBoard를 생성하면 될 것 같습니다.) game_name 역시 게임별로 고유한 값이므로 게임 이름을 외래키로 사용하였습니다.